scripts/tools/metricProgrammable.yml

```diff
@@ -0,0 +1,513 @@
+      desc: "[out] metric programmable description"
+    - type: char
+      name: "component[$T_MAX_METRIC_PROGRAMMABLE_COMPONENT_EXP]"
+      desc: "[out] metric programmable component"
+    - type: uint32_t
+      name: "tierNumber"
+      desc: "[out] tier number"
+    - type: uint32_t
+      name: "domain"
+      desc: "[out] metric domain number."
+    - type: uint32_t
+      name: "parameterCount"
+      desc: "[out] number of parameters in the programmable"
+    - type: "$t_metric_group_sampling_type_flags_t"
+      name: "samplingType"
+      desc: |
+            [out] metric sampling type.
+            returns a combination of $t_metric_group_sampling_type_flags_t.
+    - type: uint32_t
+      name: "sourceId"
+      desc: "[out] unique metric source identifier(within platform)to identify the HW block where the metric is collected."
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Metric Programmable Parameter types"
+version: "1.8"
+class: $tMetricProgrammable
+name: $t_metric_programmable_param_type_exp_t
+etors:
+    - name: ETOR_DISAGGREGATION
+      desc: "Metric is disaggregated."
+    - name: ETOR_LATENCY
+      desc: "Metric for latency measurement."
+    - name: ETOR_NORMALIZATION_UTILIZATION
+      desc: "Produces normalization in percent using raw_metric * 100 / cycles / HW instance_count."
+    - name: ETOR_NORMALIZATION_AVERAGE
+      desc: "Produces normalization using raw_metric / HW instance_count."
+    - name: ETOR_NORMALIZATION_RATE
+      desc: "Produces normalization average using raw_metric / timestamp."
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Supported value info types"
+version: "1.8"
+name: $t_value_info_type_exp_t
+etors:
+    - name: ETOR_UINT32
+      desc: "32-bit unsigned-integer"
+    - name: ETOR_UINT64
+      desc: "64-bit unsigned-integer"
+    - name: ETOR_FLOAT32
+      desc: "32-bit floating-point"
+    - name: ETOR_FLOAT64
+      desc: "64-bit floating-point"
+    - name: ETOR_BOOL8
+      desc: "8-bit boolean"
+    - name: ETOR_CSTRING
+      desc: "C string"
+    - name: ETOR_UINT8
+      desc: "8-bit unsigned-integer"
+    - name: ETOR_UINT16
+      desc: "16-bit unsigned-integer"
+    - name: ETOR_UINT64_RANGE
+      desc: "64-bit unsigned-integer range (minimum and maximum)"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Value info of type uint64_t range"
+version: "1.8"
+name: $t_value_uint64_range_exp_t
+members:
+    - type: uint64_t
+      name: "ui64Min"
+      desc: "[out] minimum value of the range"
+    - type: uint64_t
+      name: "ui64Max"
+      desc: "[out] max value of the range"
+--- #--------------------------------------------------------------------------
+type: union
+desc: "Union of value information"
+version: "1.8"
+name: $t_value_info_exp_t
+base: $t_base_desc_t
```

Comment:
Do the scripts allow for deriving a union from a base descriptor type? If so then that is a bug in the scripts.

Reply:
I dont see an error while running.
I removed the base type as well